src/Meilisearch/Index.cs

```diff
@@ -134,6 +146,16 @@ public async Task<UpdateStatus> DeleteOneDocument(string documentId)
             return await httpresponse.Content.ReadFromJsonAsync<UpdateStatus>();
         }
 
+        /// <summary>
+        /// Delete one document by its ID
+        /// </summary>
+        /// <param name="documentId">document ID</param>
+        /// <returns>Update Status with ID to look for document.</returns>
+        public async Task<UpdateStatus> DeleteOneDocument(int documentId)
+        {
```

Comment:
@barbados-clemens Awesome work on the PR.  Just want to understand your perspective on a different approach.  
What if we create a class class `Identity` which has to implement an `implicit `C# operator which can convert both Int and string. So we don't need two methods.
More details here  : https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/user-defined-conversion-operators

Reply:
I'm not familiar with with implicit/explicit operators, but it sounds promising 
> Implicit conversions don't require special syntax to be invoked and can occur in a variety of situations, for example, in assignments and methods invocations

What would that look like in practice?

if I understand the documentation correctly, Their example just wraps the type `byte` and We'd need to wrap int and string? Does this mean the sdk methods would take `Identity` instead of an int or string?
i.e. 
` public async Task<T> GetDocument<T>(Identity documentId)`

What would our Identity class look like? I think that's the part I'm unsure about since I'm not understanding exactly how the provided example would translate in this case.

Reply:
That's the wrong suggestion from my side. Just found out that we cannot add an implicit operator on the Framework defined class. 
https://stackoverflow.com/questions/17785487/implicit-operator-for-int